source/lac/scalapack.cc

```diff
@@ -943,69 +956,104 @@ ScaLAPACKMatrix<NumberType>::invert()
   const bool is_symmetric = (property == LAPACKSupport::symmetric ||
                              state == LAPACKSupport::State::cholesky);
 
-  // Matrix is neither in Cholesky nor LU state.
-  // Compute the required factorizations based on the property of the matrix.
-  if (!(state == LAPACKSupport::State::lu ||
-        state == LAPACKSupport::State::cholesky))
-    {
-      if (is_symmetric)
-        compute_cholesky_factorization();
-      else
-        compute_lu_factorization();
-    }
-  if (grid->mpi_process_is_active)
-    {
-      int         info  = 0;
-      NumberType *A_loc = &this->values[0];
+  // Check whether matrix is triangular and contents is actually a matrix
+  // or inverse of a matrix.
+  const bool is_triangular = (property == LAPACKSupport::upper_triangular ||
+                              property == LAPACKSupport::lower_triangular) &&
+                             (state == LAPACKSupport::State::matrix ||
+                              state == LAPACKSupport::State::inverse_matrix);
 
-      if (is_symmetric)
+  if (is_triangular)
+    {
+      if (grid->mpi_process_is_active)
         {
-          ppotri(&uplo,
+          const char uploTriangular =
+            property == LAPACKSupport::upper_triangular ? 'U' : 'L';
+          const char  diag  = 'N';
+          int         info  = 0;
+          NumberType *A_loc = &this->values[0];
+          ptrtri(&uploTriangular,
+                 &diag,
                  &n_columns,
                  A_loc,
                  &submatrix_row,
                  &submatrix_column,
                  descriptor,
                  &info);
-          AssertThrow(info == 0, LAPACKSupport::ExcErrorCode("ppotri", info));
+          AssertThrow(info == 0, LAPACKSupport::ExcErrorCode("ptrtri", info));
```

Comment:
is inversion stored in the same part as the matrix? If so, add a comment like

> The inversion is stored in the same part as the triangular matrix, 
> so we don't need to re-set the property here.

Reply:
Yes the inversion is stored in the same part as the matrix. I will add the above comment